src/App.tsx

```diff
@@ -15,49 +17,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Name = 'name',
+  Length = 'length',
+  None = '',
+}
+
+export const Good: React.FC<{ goods: string[] }> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good" key={good}>
+        {good}
+      </li>
+    ))}
+  </ul>
+);
+
+function getGoods(goods: string[], sortField: string, isReversed: boolean) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Name:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const toggleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const visibleGoods = getGoods(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.Name,
+          })}
+          onClick={() => {
+            setSortField(SortType.Name);
+            setIsReversed(false);
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here